src/lib/coaler/embedder/ConformerEmbedder.cpp

```diff
@@ -89,4 +137,118 @@ namespace coaler::embedder {
+                                  substructMatchParams.maxMatches, substructMatchParams.numThreads);
+
+            std::vector<RDKit::MatchVectType> targetMatches;
+            RDKit::SubstructMatch(targetMol, *mcsResult.QueryMol, targetMatches, substructMatchParams.uniquify, false,
+                                  substructMatchParams.useChirality, substructMatchParams.useQueryQueryMatches,
+                                  substructMatchParams.maxMatches, substructMatchParams.numThreads);
+
+            if (targetMatches.empty() || ligandMatches.empty()) {
+                spdlog::warn("unable to match mcs to target or ligand");
+                spdlog::warn("ligand: {}", RDKit::MolToSmiles(*worstLigandMol));
+                spdlog::warn("target: {}", RDKit::MolToSmiles(target.getMolecule()));
+                spdlog::warn("mcs: {}", mcsResult.SmartsString);
+                // continue;
+
+                // try again with strict mcs rules
+                mcsResult = get_mcs(mols, true);
+                RDKit::SubstructMatch(targetMol, *mcsResult.QueryMol, targetMatches, substructMatchParams.uniquify,
+                                      false, substructMatchParams.useChirality,
+                                      substructMatchParams.useQueryQueryMatches, substructMatchParams.maxMatches,
+                                      substructMatchParams.numThreads);
+                RDKit::SubstructMatch(*worstLigandMol, *mcsResult.QueryMol, ligandMatches,
+                                      substructMatchParams.uniquify, false, substructMatchParams.useChirality,
+                                      substructMatchParams.useQueryQueryMatches, substructMatchParams.maxMatches,
+                                      substructMatchParams.numThreads);
+            }
+
+            // if still no match in both, skip target
+            if (targetMatches.empty() || ligandMatches.empty()) {
+                continue;
+            }
+            const RDKit::MatchVectType ligandMatch = ligandMatches.at(0);
+            const RDKit::MatchVectType targetMatch = targetMatches.at(0);
+
+            const CoreAtomMapping ligandMcsCoords
+                = getLigandMcsAtomCoordsFromTargetMatch(targetConformer.getPositions(), ligandMatch, targetMatch);
+
+            RDKit::DGeomHelpers::EmbedParameters params;
+            params = RDKit::DGeomHelpers::srETKDGv3;
+            params.optimizerForceTol = forceTol;
+            params.randomSeed = seed;
+            params.coordMap = &ligandMcsCoords;
+            params.useRandomCoords = true;
+            params.numThreads = 1;
+            params.clearConfs = false;
+            int addedID = 0;
+            try {
+                addedID = RDKit::DGeomHelpers::EmbedMolecule(*worstLigandMol, params);
+            } catch (const std::runtime_error &e) {
+                spdlog::warn(e.what());
+                addedID = -1;
+            }
+            if (addedID < 0) {
```

Comment:
why not just put this block in the catch clause and print the error together with in which mol it occured?